### DISCUSSION TITLE
Update Bound Text Field When Related Record Name Changes

### DISCUSSION THREAD

**Discussion by matthewdevaney** [khoait/DCE.PCF#46] [General]
How can I update the bound text field when the related record display name changes?

Example:
1. Assume there are two tables having a many to many relationship
- Sales Leads
- Markets

2. I create a new Sales Lead and associate it to two Markets using the Polylookup control:
- California 
- New York

3. The result is this text string being written to the Markets Bound Field in the Sales Leads table
- California,New York

4. Now lets say that the display name for California gets updated to the name California2 in the Market table

5. When the display name is updated the Sales Lead record bound field does not update to California2 immediately. I must open the Sales Lead form to see the changes reflected.

Question: how can I ensure the bound field in the Sales Lead table is updated for any associated Markets when their display name changes? I do not want to open the Sales Lead form manually for each record to accomplish this.

I'm a huge fan of this component. The UI is so simple and elegant. My users love it.  Thank you so much @khoait 😀

**Comment by khoait** [comment]
Hi @matthewdevaney. Because a PCF component only runs at front end, it only be refreshed when you open the form. For background update scenarios you'll need to build your own logic using a flow or a plugin. It would be out of scope of the component.

Any idea/suggestion, please let me know. Thanks for your interest in the component, please give the repo a star if possible!

**Comment by matthewdevaney** [comment]
@khoait 
I'm happy to give you a star. Done! ⭐

Your explanation makes sense. Was simply looking for some inspiration to build for myself. It seems a flow or plugin is the correct way to go.

Is my requirement something your clients have asked for?  Have you ever built it for yourself?  I am not asking you to build this for me. I just would like to know if it's been done.

Many thanks 😊

**Reply by khoait** [reply]
Yes, I have done a plugin for this requirement previously. The plugin is triggered on Name changed, also when the option is deleted in the background with same logic. It should be fairly straightforward, let me know if you need some help.

**Reply by MooseX** [reply]
> @khoait I'm happy to give you a star. Done! ⭐
> 
> Your explanation makes sense. Was simply looking for some inspiration to build for myself. It seems a flow or plugin is the correct way to go.
> 
> Is my requirement something your clients have asked for? Have you ever built it for yourself? I am not asking you to build this for me. I just would like to know if it's been done.
> 
> Many thanks 😊

Hi @matthewdevaney - were you able to find a workaround / figure out a flow to trigger this? TIA! 

@khoait - thanks for the awesome component!

**Reply by khoait** [reply]
Hi @MooseX please find my sample plugin here
 https://github.com/khoait/DCE.PCF/blob/main/Samples%2FDCE.PCF%2FDCE.PCF.Plugins%2FAssociatePolyLookup.cs

**Reply by MooseX** [reply]
> Hi @MooseX please find my sample plugin here https://github.com/khoait/DCE.PCF/blob/main/Samples%2FDCE.PCF%2FDCE.PCF.Plugins%2FAssociatePolyLookup.cs

Thanks @khoait - will give it a try. Much appreciated.
